lib/companies_web/live/company.ex

```diff
@@ -0,0 +1,79 @@
+defmodule CompaniesWeb.CompanyLive do
+  @moduledoc false
+  use Phoenix.LiveView, layout: {CompaniesWeb.LayoutView, "live.html"}
+
+  alias CompaniesWeb.CompanyView
+  alias Companies.Accounts
+  alias Companies.Industries
+  alias Companies.Companies
+
+  alias CompaniesWeb.Router.Helpers, as: Routes
+
+  def render(assigns) do
+    Phoenix.View.render(CompanyView, "index.html", assigns)
+  end
+
+  def mount(params, session, socket) do
+    industries = Industries.for_select()
+
+    socket =
+      socket
+      |> assign_new(:current_user, fn -> Accounts.get(session["user_id"]) end)
+      |> assign(
+        user_id: session["user_id"],
+        industries: industries,
+        locale: params["locale"]
+      )
+      |> search(params)
+
+    {:ok, socket, temporary_assigns: [companies: []]}
+  end
+
+  def handle_params(_params, _uri, socket) do
+    {:noreply, socket}
+  end
+
+  def handle_event("search", params, %{assigns: assigns} = socket) do
+    socket =
+      socket
+      |> search(params)
+      |> push_patch(to: Routes.live_path(socket, CompaniesWeb.CompanyLive, assigns.locale, params))
+
+    {:noreply, socket}
+  end
+
+  def handle_event("load_more", _, %{assigns: assigns} = socket) do
+    page = assigns.page + 1
+
+    if last_page?(assigns) do
+      {:noreply, socket}
+    else
+      companies = Companies.all(%{"page" => page})
+
+      {:noreply,
+       socket
+       |> assign(companies: companies)
+       |> assign(page: page)
+       |> assign(update: "append")
+       |> push_patch(to: Routes.live_path(socket, CompaniesWeb.CompanyLive, assigns.locale))}
+    end
+  end
+
+  defp search(socket, params) do
+    companies = Companies.all(params)
+
+    socket
+    |> assign(
+      companies: companies,
+      page: 1,
+      total_pages: companies.total_pages,
+      text: params["search"]["text"],
+      industry_id: params["search"]["industry_id"]
+    )
+    |> assign(update: "replace")
```

Comment:
```elixir
assign(socket,
  companies: companies,
  page: 1,
  total_pages: companies.total_pages,
  text: params["search"]["text"],
  industry_id: params["search"]["industry_id"],
  update: "replace"
)
```
Make it a signle assign call maybe?

Reply:
Sure, covered it in this commit https://github.com/beam-community/elixir-companies/commit/9e8407437513794a3de0cc6daa03e5221e498b47 but it's not showing up in the PR